pkg/spire/spireworkload.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spire
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/pem"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	spireconfig "github.com/tektoncd/pipeline/pkg/spire/config"
+)
+
+type SpireWorkloadApiClient struct {
```

Comment:
I'm assuming that this is for the Entrypointer, and the other struct is for the controller, maybe we should change the naming to be `SpireEntrypointerApiClient` and `SpireControllerApiClient`, else it's a little confusing cause the other client includes the workload api client as well.

There's some code re-use here where the controller client can use the verification client here as well.

Reply:
Yes I agree with the name change. But code re-use? Can you point that out and I will get rid of the re-used code.

Reply:
I think it was some similarities in the dial code and the getxsvid, but i see there's some distinct differences there. I feel like anything that deals with signing and verification uses the same parts from the other struct, and the other struct is mainly for talking to the SPIRE server for entry registration.

Reply:
But that's just one opinion, think the controller and entrypointer segregation is also good.

Reply:
Move registrar on its own. Move shared code into its own file.

Reply:
change struct name and moved verify and sign into its own files.